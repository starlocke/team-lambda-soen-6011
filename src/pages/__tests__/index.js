import React from "react"
import renderer from "react-test-renderer"
import IndexPage from "../index"
import { StaticQuery, useStaticQuery } from "gatsby"
beforeEach(() => {
  console.error = jest.fn();
    StaticQuery.mockImplementationOnce(({ render }) =>
      render({
        site: {
          siteMetadata: {
            title: `Default Starter`,
          },
        },
      })
    )
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          author: 'Florian',
          description: 'My description',
          title: 'My Title',
        },
      },
    }));
})

describe("IndexPage", () => {
  it("renders correctly", () => {
    const data = {
          "main": {
            "edges": [
              {
                "node": {
                  "id": "a347f53e-43d8-5e21-8e35-3175bb099ff0",
                  "excerpt": "Classification of Skill Hard skill Definition 1 – “Because, it is technical knowledge or training that you have gained through any life experience, including in your career or education.” Definition 2 - “Hard skills are learned abilities acquired and…",
                  "frontmatter": {
                    "slug": "/skills/software-construction/tdd-testing",
                    "title": "Test Driven Development Testing",
                    "author": "Hardik Vora"
                  }
                }
              },
              {
                "node": {
                  "id": "96d58885-9dd3-57ca-8697-920bacc3a1cc",
                  "excerpt": "Classification of Skill Technical Skills: 6 Human Relation Skills: 2 Conceptual Skills: 2 RESTful API Programming & Design requires a blend of different types of skills Prerequisites for Skill Ability to query a database using SQL and an overall in…",
                  "frontmatter": {
                    "slug": "/skills/computing-foundations/restful-api-programming",
                    "title": "RESTful API Programming",
                    "author": "Warren White"
                  }
                }
            }
            ]
          },
          "category": {
            "group": [
              {
                "fieldValue": "computing-foundations",
                "totalCount": 1
              },
              {
                "fieldValue": "software-configuration-management",
                "totalCount": 1
              },
            ]
          },
          "author": {
            "group": [
              {
                "fieldValue": "Chirag Vora",
                "totalCount": 1
              },
              {
                "fieldValue": "Hao Zhang",
                "totalCount": 1
              },
              
            ]
          }
        }
    
      
    const tree = renderer.create(<IndexPage data={data} />).toJSON();
    expect(tree).toMatchSnapshot()
  })
})