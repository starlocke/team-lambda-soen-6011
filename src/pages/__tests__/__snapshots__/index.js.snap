// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Default Starter
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <div>
      <div
        className="search__container"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <label
          htmlFor="searchInput"
        >
          Search All Skill Pages
        </label>
        <input
          autoComplete="off"
          className="search__input"
          id="searchInput"
          onChange={[Function]}
          placeholder="Query"
          type="text"
          value=""
        />
        <div
          className=""
        >
          
        </div>
      </div>
      <noscript>
        <form
          action="https://www.google.com/search"
          method="GET"
        >
          <input
            name="q"
            type="hidden"
            value="site:sestopia-gatsby.herokuapp.com"
          />
          <label>
            Search Sestopia:
            <br />
            <input
              name="q"
              placeholder="Query"
              type="text"
              value=""
            />
          </label>
        </form>
      </noscript>
    </div>
    <main>
      <div>
        <h2>
          Categories
        </h2>
        <ul
          className="category__wrapper"
        >
          <li>
            <a
              href="/categories/computing-foundations/"
            >
              computing-foundations
               (
              1
              )
            </a>
          </li>
          <li>
            <a
              href="/categories/software-configuration-management/"
            >
              software-configuration-management
               (
              1
              )
            </a>
          </li>
        </ul>
      </div>
      <div>
        <h2>
          Authors
        </h2>
        <ul
          className="author__wrapper"
        >
          <li>
            <a
              href="/author/chirag-vora/"
            >
              Chirag Vora
               (
              1
              )
            </a>
          </li>
          <li>
            <a
              href="/author/hao-zhang/"
            >
              Hao Zhang
               (
              1
              )
            </a>
          </li>
        </ul>
      </div>
      <div>
        <div
          style={
            Object {
              "margin": "2.5em 0",
            }
          }
        >
          <a
            href="/skills/software-construction/tdd-testing"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          >
            Test Driven Development Testing
          </a>
          <div>
            <em>
              by 
              Hardik Vora
            </em>
          </div>
          <div>
            Classification of Skill Hard skill Definition 1 – “Because, it is technical knowledge or training that you have gained through any life experience, including in your career or education.” Definition 2 - “Hard skills are learned abilities acquired and…
          </div>
        </div>
        <div
          style={
            Object {
              "margin": "2.5em 0",
            }
          }
        >
          <a
            href="/skills/computing-foundations/restful-api-programming"
            style={
              Object {
                "fontSize": "1.2em",
              }
            }
          >
            RESTful API Programming
          </a>
          <div>
            <em>
              by 
              Warren White
            </em>
          </div>
          <div>
            Classification of Skill Technical Skills: 6 Human Relation Skills: 2 Conceptual Skills: 2 RESTful API Programming & Design requires a blend of different types of skills Prerequisites for Skill Ability to query a database using SQL and an overall in…
          </div>
        </div>
      </div>
      <footer
        className="sc-AxjAm ipahVA"
      >
        SESTOPIA - An Information Intensive Application
      </footer>
    </main>
  </div>,
]
`;
